### PR TITLE
[M] CANDLEPIN-912: Updated product endpoints to support improved querying

### PR DESCRIPTION
**Pull request by Ceiu**
- Added the "name", "active", and "custom" filters to the query parameters of the GET /owners/{key}/products endpoint
- Added the "owner", "name", "active", and "custom" filters to the query parameters of the GET /products endpoint
- Removed the now-redundant "omit_globals" query parameter from both GET /owners/{key}/products and GET /products endpoints
- Reworked the GET /products and GET /owner/{key}/products endpoints to support filtering on product IDs, product names, and the "active" and "custom" states
- Added a new query builder object for building dynamic product queries

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -6076,23 +6076,93 @@ paths:
...
+            be passively included, excluded or omitted from the output, or exclusively selected
+            as the only products to return.
+          required: false
+          schema:
+            type: string
+            enum:
+              - include
+              - exclude
+              - exclusive
+            default: exclusive
```
Comment (joshmalbrecht) [inline api/candlepin-api-spec.yaml:6149]: This should be "include"

Comment (Ceiu) [reply api/candlepin-api-spec.yaml]: Fixed

File: spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java [hunk 2]
```diff
@@ -781,17 +622,895 @@ private ProductDTO createProductWithProvidedAndDerivedProduct(String ownerKey) {
...
+        @Test
+        public void shouldErrorWithInvalidActiveInclusion() {
+            OwnerDTO owner = this.owners.get(1);
+
+            assertBadRequest(() -> this.adminClient.ownerProducts()
+                .getProductsByOwner(owner.getKey(), null, null, "invalid_type", INCLUSION_INCLUDE));
+        }
+
+        @Test
+        public void shouldAllowFilteringWithCustomIncluded() {
```
Comment (bakajstep) [inline spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java:1079]: This one

**Review by bakajstep — COMMENTED** [review]
It look good to me just found 3 test which are the same so i just have question if it is ok or we want to update it?

**Review by joshmalbrecht — APPROVED** [review]

File: spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java (hunk 2 above)
Comment (Ceiu) [reply spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java:1079]: These being effectively duplicates is fine. When it comes to writing tests and verifying behavior, you don't want to combine separate validations as you end up losing that in-code contract should test implementations change in the future.

To some extent, this is along the lines of self-documenting code; but we're leaning more on proper test isolation and verifying a specification.

**Review by bakajstep — APPROVED** [review]

**Pull request merged**